ldaptor/protocols/ldap/distinguishedname.py

```diff
@@ -124,23 +110,27 @@ def __init__(self, stringValue=None, attributeType=None, value=None):
         if stringValue is None:
             assert attributeType is not None
             assert value is not None
-            self.attributeType = attributeType
-            self.value = value
+            self.attributeType = _to_unicode(attributeType)
+            self.value = _to_unicode(value)
         else:
             assert attributeType is None
             assert value is None
 
-            if isinstance(stringValue, bytes):
-                equal_char = b'='
-            else:
-                equal_char = '='
+            stringValue = _to_unicode(stringValue)
 
-            if equal_char not in stringValue:
+            if '=' not in stringValue:
+                if six.PY2:
+                    stringValue = _to_bytes(stringValue)
                 raise InvalidRelativeDistinguishedName(stringValue)
-            self.attributeType, self.value = stringValue.split(equal_char, 1)
+            self.attributeType, self.value = stringValue.split('=', 1)
 
     def __str__(self):
-        return '='.join((escape(self.attributeType), escape(self.value)))
+        if not six.PY2:
+            return self.__bytes__().decode('utf-8')
+        return self.__bytes__()
```

Comment:
I think that we should not return bytes here, but a real human readable representation of this object.
In this way, we can make sure that any usage of `__str__` with the intended purpose of getting the wire representation will fail